frontend/packages/dev-console/src/components/pipelines/pipeline-form/PipelineSecretSection.tsx

```diff
@@ -0,0 +1,100 @@
+import * as React from 'react';
+import { Formik } from 'formik';
+import { PlusCircleIcon } from '@patternfly/react-icons';
+import { Button } from '@patternfly/react-core';
+import { ExpandCollapse } from '@console/internal/components/utils';
+import { SecretType } from '@console/internal/components/secrets/create-secret';
+import { SecretModel } from '@console/internal/models';
+import { k8sCreate } from '@console/internal/module/k8s';
+import SecretForm from './SecretForm';
+import {
+  associateServiceAccountToSecret,
+  getTektonSecretAnnotations,
+} from '../../../utils/pipeline-utils';
+import SecretsList from './SecretsList';
+import { advancedSectionValidationSchema } from './pipelineForm-validation-utils';
+import { SecretAnnotationId } from '../const';
+
+import './PipelineSecretSection.scss';
+
+const initialValues = {
+  secretName: '',
+  annotations: { key: SecretAnnotationId.Image, value: '' },
+  type: SecretType.dockerconfigjson,
+  formData: {},
+};
+
+type PipelineSecretSectionProps = {
+  namespace: string;
+};
+
+const PipelineSecretSection: React.FC<PipelineSecretSectionProps> = ({ namespace }) => {
+  const [addSecret, setAddSecret] = React.useState(false);
+
+  const handleSubmit = (values, actions) => {
+    actions.setSubmitting(true);
+    const newSecret = {
+      apiVersion: SecretModel.apiVersion,
+      kind: SecretModel.kind,
+      metadata: {
+        name: values.secretName,
+        namespace,
+        annotations: getTektonSecretAnnotations(values.annotations),
+      },
+      type: values.type,
+      stringData: values.formData,
+    };
+    k8sCreate(SecretModel, newSecret)
+      .then((resp) => {
+        actions.setSubmitting(false);
+        setAddSecret(false);
+        associateServiceAccountToSecret(resp, namespace);
+      })
+      .catch((err) => {
+        actions.setSubmitting(false);
+        actions.setStatus({ submitError: err.message });
+      });
+  };
+
+  const handleReset = (values, actions) => {
+    actions.resetForm({ values: initialValues, status: {} });
+    setAddSecret(false);
+  };
+
+  const handleAddSecret = () => {
```

Comment:
why do we need this functon?

Reply:
This callback is used to set state which in turn conditionally render the add button/secret form. These changes are from PR https://github.com/openshift/console/pull/4868 cc: @vikram-raj

Reply:
Yes, this function is for showing and hiding the Create secret form.

Reply:
What I mean is there is just one line inside the function, isn't it better to use directly, like
`(() =>oneline)() `

Just to confirm, if it conforms to the use cases..

Reply:
So, when I reviewed it was a one-liner so, I thought we need to call it directly...

Reply:
Based on my last review... recommend just inlining the onClick set value. No need for the extra method.